[SE-0382] Stage in prototypes of macros defined in the standard library.

DougGregor opened the pull request:
These are inactive at the moment; we'll conditionally take advantage of them later on. Part of rdar://104346457.

DougGregor committed:
[SE-0382] Stage in prototypes of macros defined in the standard library.

This are inactive at the moment; we'll conditionally take advantage of them
later on. Part of rdar://104346457.

DougGregor commented:
@swift-ci please smoke test

rxwei commented on stdlib/public/core/Macros.swift:
```diff
@@ -24,4 +23,40 @@
 public macro externalMacro<T>(module: String, type: String) -> T =
     Builtin.ExternalMacro
```
Trivial nit: Inconsistent indentation with the rest

```suggestion
  Builtin.ExternalMacro
```

DougGregor committed:
Indentation fix

DougGregor committed:
[Macros] Drop special code completions for built-in macros.

We don't need special completion logic for things like `#file` and
`#line` now that they are declared in the standard library. Drop it
and update tests.

DougGregor commented:
@swift-ci please smoke test

DougGregor committed:
Update tests

DougGregor commented:
@swift-ci please smoke test
